classes/class-kss-compare-totals.php

```diff
@@ -0,0 +1,44 @@
+<?php // phpcs:ignore
+/**
+ * Free orders class
+ *
+ * @package KlarnaShippingService/Classes
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Free Orders class
+ */
+class KSS_Compare_Totals {
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		add_action( 'kco_wc_process_payment', array( $this, 'compare_kco_kss_totals' ), 10, 2 );
+	}
+
+	/**
+	 * Kompare KCO and KSA order shipping totals for uncaptured amounts and warn the customer.
+	 *
+	 * @param int   $order_id The WooCommerce order ID.
+	 * @param array $klarna_order The Klarna order.
+	 * @return void
+	 */
+	public function compare_kco_kss_totals( $order_id, $klarna_order ) {
+
+		$order = wc_get_order( $order_id );
+
+		$klarna_total_amount = round( $klarna_order['selected_shipping_option']['price'] / 100, 2 );
+		$order_total_amount  = ( $order->get_shipping_total() + $order->get_shipping_tax() );
+
+		if ( $klarna_total_amount > $order_total_amount && ( $klarna_total_amount - $order_total_amount ) > 3 ) {
```

Comment:
This currently only catches if Klarna amount is bigger then the order amount. Technically here you will allow a klarna order for 0 and woo order for 1000 to go through.

I think a nicer way to do this would be to do something like this:
```
$dif = $klarna_total_amount - $order_total_amount;

if( $dif > 3 || $dif < -3 ) {
    $order->add_order_note(...)
}
```

Since its a shorter if statement its easier to read at a glance, and its easier to debug. Also you test both cases of Klarna being more expensive, and being cheaper then what should be the case

Reply:
@mntzrr Keep it as the suggestion i wrote, since its more legible then using absint at a glance, and the cost of running the extra check is minimal.